File: 04-deberes/02-proyecto/Backend/zapatos/api/models/Zapato.js
/**
 * Zapato.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
    marca: {
      type: 'string',
      required: true,
      minLength: 4,
      maxLength: 15,
    },
    color: {
      type: 'string',
      required: true,
      minLength: 6,
      maxLength: 15,
    },
    talla: {
      type: 'number',
      required: true,
      min: 25,
      max: 42,
    },
    tipo: {
      type: 'string',
      required: true,
      unique: true,
      maxLength: 6,
    },
    cantidad: {
      type: 'number',
      required: true,
      min: 0,
    },
    precio: {
      type: 'number',
      required: true,
      min: 0,
    },
    compraDeZapato: {     // Nombre atributo de la relación
      collection: 'compra', // Nombre del modelo a relacionar
      via: 'codigoZap'        // Nombre del campo a hacer la relacion
    },
  },
}

File: 04-deberes/02-proyecto/AppZapatos/app/src/main/java/com/example/appzapatos/modulo_zapatos/Zapato.kt
package com.example.appzapatos.modulo_zapatos

import com.example.appzapatos.modulo_compras.Compra
import java.util.*
import kotlin.collections.ArrayList

class Zapato(
    var compraDeZapato: ArrayList<Compra>?,
    var createdAt: Long?,
    var updatedAt: Long?,
    var id: Int?,
    var marca: String,
    var color: String,
    var talla: Int,
    var tipo: String,
    var cantidad: Int,
    var precio: Double
) {

}